raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -226,7 +221,7 @@ public void initialize(OffsetAndEpoch logEndOffsetAndEpoch) throws IllegalStateE
                     time,
                     election.epoch(),
                     OptionalInt.of(election.leaderId()),
-                    Optional.empty(),
+                    election.optionalVotedKey(),
```

Comment:
it might not be expected that a state has both leader and voted key state, but I think it's better not to lose the state if it does happen to exist.

not an issue for backwards compatibility, older version would lose votedkey state but would have leader state and will reject standard votes correctly because of that.